implementations/rust/ockam/ockam/src/system/handler.rs

```diff
@@ -1,17 +1,47 @@
-use ockam_core::compat::boxed::Box;
-use ockam_core::{Message, Result, Routed};
+use dyn_clonable::*;
+use dyn_clone::DynClone;
+use ockam_core::compat::{boxed::Box, collections::BTreeMap, string::String};
+use ockam_core::{Address, Message, Result, Routed};
 
 /// Handle a single type of message for a worker system-address
 ///
 /// A handle may re-emit messages to the worker system, or to the
 /// Ockam runtime.  All state associated with a particular protocol
 /// must be contained in the type that implements this trait.
+#[clonable]
```

Comment:
I'm familiar with `dyn_clone` (and have no issues with our use of it), but not `dyn_clonable`.

What does this save us? If it's just nicer syntax than `dyn_clone::clone_trait_object!(SystemHandler);` or whatever, I'm not sure this is worth an additional dependency, let alone on a proc macro (which have a much large impact on build times than library crates).

If it's more significant than perhaps it's fine, though

Reply:
This is probably the only thing I'd like to change if possible before landing.

Reply:
`dyn_clone::clone_trait_object!(SystemHandler); ` in this context is extremely hard to use because of the generic type bounds on `SystemHandler`